tests/components/whirlpool/test_climate.py

```diff
@@ -0,0 +1,236 @@
+    assert aircon_entity.max_temp == 30
+
+
+async def test_get_attributes(aircon_entity: AirConEntity, aircon_api):
+    """Test getting attributes."""
+    aircon_api.return_value.get_online.return_value = False
+    aircon_api.return_value.get_power_on.return_value = False
+    assert not aircon_entity.available
+    assert aircon_entity.hvac_mode == HVAC_MODE_OFF
+
+    aircon_api.return_value.get_online.return_value = True
+    assert aircon_entity.available
+    assert aircon_entity.hvac_mode == HVAC_MODE_OFF
+
+    aircon_api.return_value.get_power_on.return_value = True
+
+    aircon_api.return_value.get_mode.return_value = whirlpool.aircon.Mode.Cool
+    assert aircon_entity.hvac_mode == HVAC_MODE_COOL
+    aircon_api.return_value.get_mode.return_value = whirlpool.aircon.Mode.Heat
+    assert aircon_entity.hvac_mode == HVAC_MODE_HEAT
+    aircon_api.return_value.get_mode.return_value = whirlpool.aircon.Mode.Fan
+    assert aircon_entity.hvac_mode == HVAC_MODE_FAN_ONLY
+
+    aircon_api.return_value.get_fanspeed.return_value = whirlpool.aircon.FanSpeed.Auto
+    assert aircon_entity.fan_mode == FAN_AUTO
+    aircon_api.return_value.get_fanspeed.return_value = whirlpool.aircon.FanSpeed.Low
+    assert aircon_entity.fan_mode == FAN_LOW
+    aircon_api.return_value.get_fanspeed.return_value = whirlpool.aircon.FanSpeed.Medium
+    assert aircon_entity.fan_mode == FAN_MEDIUM
+    aircon_api.return_value.get_fanspeed.return_value = whirlpool.aircon.FanSpeed.High
+    assert aircon_entity.fan_mode == FAN_HIGH
+    aircon_api.return_value.get_fanspeed.return_value = whirlpool.aircon.FanSpeed.Off
+    assert aircon_entity.fan_mode == FAN_OFF
+
+    aircon_api.return_value.get_current_temp.return_value = 15
+    aircon_api.return_value.get_temp.return_value = 20
+    aircon_api.return_value.get_current_humidity.return_value = 80
+    aircon_api.return_value.get_humidity.return_value = 50
+    aircon_api.return_value.get_h_louver_swing.return_value = True
+    assert aircon_entity.current_temperature == 15
+    assert aircon_entity.target_temperature == 20
+    assert aircon_entity.current_humidity == 80
+    assert aircon_entity.target_humidity == 50
+    assert aircon_entity.swing_mode == SWING_HORIZONTAL
+
+    aircon_api.return_value.get_current_temp.return_value = 16
+    aircon_api.return_value.get_temp.return_value = 21
+    aircon_api.return_value.get_current_humidity.return_value = 70
+    aircon_api.return_value.get_humidity.return_value = 55
+    aircon_api.return_value.get_h_louver_swing.return_value = False
+    assert aircon_entity.current_temperature == 16
+    assert aircon_entity.target_temperature == 21
+    assert aircon_entity.current_humidity == 70
+    assert aircon_entity.target_humidity == 55
+    assert aircon_entity.swing_mode == SWING_OFF
+
+
+async def test_set_attributes(aircon_entity: AirConEntity, aircon_api):
+    """Test setting attributes."""
+    aircon_api.return_value.set_temp = AsyncMock()
```

Comment:
Setup the integration and check the values in the state machine using `hass.states.get`

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

Reply:
done